### PR TITLE
Refactor 1

### PR DESCRIPTION
**Pull request by calebrs**
- Made flags and key into a class with the `FlagManager` and `KeyManager` classes.
- Removed unneeded comments
- Added a validator middleware that checks to make sure the webhook is from the ff manager

### CODE REVIEW

File: controllers/flagsController.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-const { setFlags, returnFlags } = require("../lib/flags.js");
+const { flagManager } = require("../lib/flags.js");
 const { sendToClients } = require("./streamController");
 
 async function replaceFlags(req, res, next) {
   try {
-    setFlags(req.body);
+    flagManager.setFlags(req.body);
```
Comment (seanrichardson95) [inline controllers/flagsController.js:6]: I like how you made this into an object

File: lib/flags.js [hunk 2]
```diff
@@ -1,27 +1,29 @@
 const axios = require("axios");
 require("dotenv").config();
 
-let flags = [];
+class FlagManager {
+  constructor() {
+    this._flags = [];
```
Comment (seanrichardson95) [inline lib/flags.js:6]: Any particular reason for using an `_` here? No problems with it, just wondering

File: lib/validateWebhook.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require("dotenv").config();
+
+function validateKWebhook(req, res, next) {
```
Comment (seanrichardson95) [inline lib/validateWebhook.js]: What's with the "K" here?

File: lib/validateWebhook.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  const validateHeader = req.headers["webhook-validator"];
+
+  if (process.env.WEBHOOK_VALIDATOR !== validateHeader) {
```
Comment (seanrichardson95) [inline lib/validateWebhook.js:6]: What's saved in WEBHOOK_VALIDATOR? The URL of the FF manager?

File: lib/validateWebhook.js (hunk 3 above)
Comment (calebrs) [reply lib/validateWebhook.js]: woops, typo

File: lib/validateWebhook.js (hunk 4 above)
Comment (calebrs) [reply lib/validateWebhook.js:6]: It's the new environment variable.

File: lib/flags.js (hunk 2 above)
Comment (calebrs) [reply lib/flags.js:6]: Yeah, I noticed it in Pioneer and have also seen it around in other places. It's a semantic thing that denotes that it's sort of private. Like whoever uses the api shouldn't be messing with variables with `_` directly, and they should be using the methods to interact with the variable.

From an article: "The use of underscores to designate secret methods is a typical trend in languages without access modifiers"

**Comment by seanrichardson95** [comment]
Looks good to me. Great job Caleb!

**Comment by julia-martin** [comment]
Great job! Looks good to me!

**Pull request merged**